### ISSUE TITLE
Make `reproject` safer across CRS

### ISSUE THREAD

**Issue by asinghvi17** [JuliaGeo/GeometryOps.jl#205]
Apparently it can error if your polygon is completely outside the valid area.  Should we make it return:
- All points NaN
- An empty geometry
- A better error
- Something else
?

cc @alex-s-gardner who brought this up on Slack

**Comment by alex-s-gardner** [comment]
@asinghvi17 what are you defining as a `valid` area? Is it just anywhere where the algorithm doesn't error?

It would be nice if reproject had a `clip_src` option... and maybe a `clip2validdomain` = true/false

**Comment by asinghvi17** [comment]
Yeah that would be nice...the problem is, that we don't actually have valid bounds :D 

https://github.com/MakieOrg/GeoMakie.jl/blob/e897be2d7227e9b935efc0657276004f51fcd138/src/projection.jl#L37-L46 is what I use to handle this in GeoMakie, it could be useful here - but it does add a bit of time to the algorithm.

**Comment by alex-s-gardner** [comment]
What about returning an error along the lines of "some points fall outside of valid projection domain, try supplying a valid extent to clipsrc"

**Comment by rafaqz** [comment]
If we just error by default but we let users pass an argument to get slower but correct/NaN results?
